xpublish/utils/zarr.py

```diff
@@ -104,10 +108,11 @@ def create_zmetadata(dataset):
     zmeta['metadata'][group_meta_key] = {'zarr_format': zarr_format}
     zmeta['metadata'][attrs_key] = _extract_dataset_zattrs(dataset)
 
-    for key, da in dataset.variables.items():
+    for key in dataset.variables.keys():
+        da = dataset[key]
         encoded_da = encode_zarr_variable(da, name=key)
         encoding = extract_zarr_variable_encoding(da)
-        zmeta['metadata'][f'{key}/{attrs_key}'] = _extract_dataarray_zattrs(encoded_da)
+        zmeta['metadata'][f'{key}/{attrs_key}'] = _extract_dataarray_zattrs(da)
```

Comment:
I think we may need to work on the `encoded_da`, as the attrs likely change within `encode_zarr_variable()` as it converts datetimes, rescales, and adds encodings.

https://github.com/pydata/xarray/blob/d4db16699f30ad1dc3e6861601247abf4ac96567/xarray/backends/zarr.py#L280-L310